network/test/middleware_test.go

```diff
@@ -491,17 +491,20 @@ func (m *MiddlewareTestSuite) TestUnsubscribe() {
 		}
 	}
 
+	msgRcvd := make(chan struct{}, 2)
+	msgRcvdFun := func() {
+		<-msgRcvd
+	}
 	message1 := createMessage(firstNode, lastNode, "hello1")
-
-	m.ov[last].On("Receive", firstNode, mockery.Anything).Return(nil).Once()
+	m.ov[last].On("Receive", firstNode, mockery.Anything).Return(nil).Run(func(_ mockery.Arguments) {
+		msgRcvd <- struct{}{}
+	})
 
 	// first test that when both nodes are subscribed to the channel, the target node receives the message
 	err := m.mws[first].Publish(message1, testChannel)
 	assert.NoError(m.T(), err)
 
-	assert.Eventually(m.T(), func() bool {
-		return m.ov[last].AssertCalled(m.T(), "Receive", firstNode, mockery.Anything)
```

Comment:
`AssertCalled` calls `Fail` if the assertion doesn't succeed. This caused the flakiness if the message was not delivered in 1 millisecond before the first call to `AssertCalled`. Subsequent calls may succeed but by then the test has already been marked as failed :(

Reply:
🤯